### PR TITLE
Add an easily extendable PageConfigurations object to replace the…

### PR DESCRIPTION
**Pull request by jkeck**
… current hard-coded data hash.

This is intended to allow apps or plugins providing SirTrevor widgets to inject their own configurations into the page.

### CODE REVIEW

**Comment by jkeck** [comment]
Not married to this pattern/approach, so happy to explore other options.

File: app/models/spotlight/page_configurations.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
+module Spotlight
+  ##
+  # PageConfigurations is a simple class to gather and return all
+  # configurations needed for the various SirTrevor widgets.
+  # A downstream application can inject its own data like so
+  # Spotlight::PageConfigurations.local_params = {
```
Comment (mejackreed) [inline app/models/spotlight/page_configurations.rb]: Where is the recommended place for this to happen in a Spotlight application?

Comment (jkeck) [reply app/models/spotlight/page_configurations.rb]: Most likely in an initializer.  Should I add that to this comment?

Comment (mejackreed) [reply app/models/spotlight/page_configurations.rb]: Ah gotcha.. sure / maybe.. i was wondering if these were like per widget configs , but I guess they are global which is fine

Comment (cbeer) [reply app/models/spotlight/page_configurations.rb]: If we're putting them in an initializer, does it make sense to add them as engine configuration, just for the sake of keeping everything in one place?

Comment (jkeck) [reply app/models/spotlight/page_configurations.rb]: Sure, that makes sense.

Comment (jkeck) [reply app/models/spotlight/page_configurations.rb]: I've updated the code and relevant comment, let me know if there should be addl. changes.

**Pull request merged**